lib/virt_autotest/virtual_network_utils.pm

```diff
@@ -362,4 +382,14 @@ sub setup_vm_simple_dns_with_ip {
     record_info("Simple DNS setup in /etc/hosts for $_ip $_vm is successful!", script_output("cat /etc/hosts"));
 }
 
+sub update_simple_dns_for_all_vm {
+    my $_vnet = shift;
+
+    my $_cmd = "virsh list --all | grep -e sles -e opensuse -e alp -i | awk \'{print \$2}\'";
+    my $_vms = script_output($_cmd, 30, type_command => 0, proceed_on_failure => 0);
+    my @_vm_array = split(/\n+/, $_vms);
+    foreach my $_vm (@_vm_array) {
+        check_guest_ip("$_vm", net => $_vnet);
+    }
```

Comment:
```
check_guest_ip($_vm, net => $_vnet) foreach my $_vm (split(/\n+/, $_vms)); 
```
is enough.

Reply:
yes, both works.

Reply:
I thin, at least, @_vm_array is not necessary.

Reply:
ok, refine them together then